extensions/git/src/repository.ts

```diff
@@ -1341,6 +1342,26 @@ export class Repository implements Disposable {
 		await this.run(Operation.RenameBranch, () => this.repository.renameBranch(name));
 	}
 
+	@throttle
+	async fastForwardBranch(name: string): Promise<void> {
+		const config = workspace.getConfiguration('git', Uri.file(this.root));
+		const fetchBeforeCheckout = config.get<boolean>('fetchBeforeCheckout', false) === true;
+
+		if (!fetchBeforeCheckout) {
+			return;
+		}
```

Comment:
Shouldn't this _config checking_ code be in `CheckoutItem.run()` instead?

Reply:
Yeah, I have considered that but I decided to push that one level below in case there will be other "entry points" to the action of fast forwarding a branch. The code feels more clean this way, but if would feel strongly about it, I can lift this up one level.

Reply:
My suggestion came specifically from the fact that there might be other entry points. It wouldn't make any sense for a `fastForwardBranch` method to not fast forward a branch, whenever `fetchBeforeCheckout` is `false`, given that fast-forwarding has nothing to do with the checkout action. The combination of both actions is something very specific to the behavior `CheckoutItem.run()`. So, `fastForwardBranch` should always fast forward a branch, despite of that setting.

Reply:
Doh! Makes sense. Lifted this up.